staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -281,16 +287,28 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 
 	<-stopCh
 
-	return s.RunPreShutdownHooks()
+	err = s.RunPreShutdownHooks()
+	if err != nil {
+		return err
+	}
+
+	// Wait for all requests to finish, which is bounded by the RequestTimeout variable.
+	s.HandlerChainWaitGroup.Wait()
```

Comment:
Just realized: this changes the shutdown behaviour of every apiserver. We now wait for running requests to finish. This can take up to 60sec. I think it's the right behaviour, but we at least have to document that in the release notes. Wondering whether we also have make this configurable, either via Config and/or flag.

@deads2k ^^

Reply:
@hzxuzhonghu I update the changelog in this direction. Can you prove-read that it is correct?

Reply:
Sure